### PR TITLE
Separate Flag to detect autogenerated Translations

### PR DESCRIPTION
**Pull request by maennchen**
Follow up from https://github.com/elixir-gettext/gettext/pull/295#issuecomment-983994697

### CODE REVIEW

File: lib/gettext/po/translations.ex [hunk 1]
```diff
@@ -29,7 +29,8 @@ defmodule Gettext.PO.Translations do
   """
   @spec autogenerated?(PO.translation()) :: boolean
   def autogenerated?(%module{flags: flags} = _translation) when is_translation(module) do
-    MapSet.member?(flags, "elixir-format")
+    # TODO: Deprecate elixir-format flag to detect autogenerated translations
+    not MapSet.disjoint?(flags, MapSet.new(["elixir-autogenerated", "elixir-format"]))
```
Comment (maennchen) [inline lib/gettext/po/translations.ex:33]: How would you like to proceed with this?

- like it is now
- remove it
- show some kind of warning

Comment (josevalim) [reply lib/gettext/po/translations.ex:33]: `MapSet.member?(flags, "elixir-format") or MapSet.member?(flags, "elixir-autogenerated")` would be my vote. We can leave a note to remove it in the future.

**Review by josevalim — COMMENTED** [review]
Looks good to me, just one tiny comment.

Actually, I am thinking `elixir-autogenerated` is a bit too long and we may have thousands of translations. Maybe `elixir-gen` or `elixir-autogen`? Or `from-elixir`?

**Comment by maennchen** [comment]
@josevalim i like the long name because it is very descriptive of what it is used for and storage is cheap. I like elixir-autogen most of your suggestions.

**Comment by josevalim** [comment]
@whatyouhide what is your pick? :)

File: lib/gettext/po/translations.ex (hunk 1 above)
Comment (maennchen) [reply lib/gettext/po/translations.ex]: @josevalim Done

**Comment by whatyouhide** [comment]
`ex-autogen` would be my pick, if we're going for shortness. Otherwise `elixir-gen` or `elixir-autogen` are both fine!

**Comment by maennchen** [comment]
@josevalim / @whatyouhide I changed it to `elixir-autogen`.

**Comment by maennchen** [comment]
@whatyouhide Should've read your comment more carefully. `ex-autogen` is also a good option. I'll change it.

File: lib/gettext/po/translations.ex [hunk 2]
```diff
@@ -29,7 +29,8 @@ defmodule Gettext.PO.Translations do
   """
   @spec autogenerated?(PO.translation()) :: boolean
   def autogenerated?(%module{flags: flags} = _translation) when is_translation(module) do
-    MapSet.member?(flags, "elixir-format")
+    # TODO: Deprecate elixir-format flag to detect autogenerated translations
```
Comment (josevalim) [inline lib/gettext/po/translations.ex:32]:
```suggestion
    # TODO: Remove elixir-format flag to detect autogenerated translations in future releases
```

**Comment by josevalim** [comment]
Just one minor comment and we can ship it.

File: lib/gettext/po/translations.ex (hunk 2 above)
Comment (maennchen) [reply lib/gettext/po/translations.ex]: Done

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**